t/integration/test_canvas.py

```diff
@@ -232,6 +233,30 @@ def test_ready_with_exception(self):
         while not result.ready():
             pass
 
+
+    @flaky
+    def test_not_ready_with_http_exception(self):
+        g = group([add.s(1, 2), raise_custom_error.s()])
+        result = g.apply_async()
+        with pytest.raises(TypeError) as e:
+            result.join_native()
+
+    @flaky
+    def test_raises_wrong_exception(self):
+        g = group([add.s(1, 2), raise_custom_error.s()])
+        result = g.apply_async()
+        with pytest.raises(urllib.error.HTTPError) as e:
+            result.join_native()
+
+
+    @flaky
+    def test_raise_correct_error(self):
+        g = group([add.s(1, 2), raise_error.s()])
+        result = g.apply_async()
+        with pytest.raises(ValueError) as e:
```

Comment:
Just so I'll get this right, this **should** fail with a ValueError right?
Instead, it raises another exception?

Reply:
No,this is correct.Just testing to see that indeed the method propagates correct the exception